src/extension-runners/chromium.js

```diff
@@ -151,12 +151,37 @@ export class ChromiumExtensionRunner {
     });
   }
 
-  wssBroadcast(data: Object) {
-    for (const client of this.wss.clients) {
-      if (client.readyState === WebSocket.OPEN) {
-        client.send(JSON.stringify(data));
+  async wssBroadcast(data: Object) {
+    return new Promise((resolve) => {
+      const clients = this.wss ? Array.from(this.wss.clients) : null;
+
+      if (!clients || clients.length === 0) {
+        resolve();
       }
-    }
+
+      const webExtReloadComplete = async (message) => {
+        const msg = JSON.parse(message);
+
+        if (msg.type === 'webExtReloadExtensionComplete') {
+          clients.forEach((client) =>
+            client.removeListener('message', webExtReloadComplete)
```

Comment:
Let's be consistent - use `off` because you used `on` before.

(the behavior is identical, but for consistency use on/off (preferred) or addListener/removeListener).